sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ContinuousScanExec.scala

```diff
@@ -47,6 +47,16 @@ case class ContinuousScanExec(
     stream.createContinuousReaderFactory()
   }
 
+  /**
+   * The callback function which is called when the output iterator of input RDD is consumed
+   * completely.
+   */
+  private def onOutputCompletion(reader: PartitionReader[_]) = {
+    reader.getCustomMetrics.foreach { metric =>
+      longMetric(metric.getName) += metric.getValue
+    }
```

Comment:
Should we also have a way to set task input metrics? Maybe a reserved metric name like `input-bytes` that will set `TaskContext.get().taskMetrics().inputMetrics.bytesRead`. Records read would be really useful as well.

Reply:
+1 will add in later implementation PRs.